gossipsub.go

```diff
@@ -564,17 +579,20 @@ func (gs *GossipConfigurableRouter) emitGossip(topic string, peers map[peer.ID]s
 		return
 	}
 
-	gpeers := gs.getPeers(topic, GossipSubD, func(peer.ID) bool { return true })
-	for _, p := range gpeers {
-		// skip mesh peers
-		_, ok := peers[p]
-		if !ok || ok {
-			gs.pushGossip(p, &pb.ControlIHave{TopicID: &topic, MessageIDs: mids})
-		}
+	emitPeers := config.GetEmitPeers(gs.wrapGetPeers(topic), peers)
```

Comment:
@vyzo adding an `EmittingStrategy` to the configuration was one way to allow developers to choose which peers messages will get rebroadcasted to.